File: stack.cpp
#include <iostream>

#define MAX 100  

class Stack {
private:
    int arr[MAX];
    int top;

public:
    Stack() {
        top = -1;
    }

    bool isEmpty() const {
        return top == -1;
    }

    bool isFull() const {
        return top == MAX - 1;
    }

    void push(int value) {
        if (isFull()) {
            std::cerr << "Stack overflow" << std::endl;
            return;
        }
        arr[++top] = value;
    }

    int pop() {
        if (isEmpty()) {
            std::cerr << "Stack underflow" << std::endl;
            return -1; 
        }
        return arr[top--];
    }

    int peek() const {
        if (isEmpty()) {
            std::cerr << "Stack is empty" << std::endl;
            return -1;  
        }
        return arr[top];
    }
};

int main() {
    Stack s;
    s.push(10);
    s.push(20);
    std::cout << "Top element is: " << s.peek() << std::endl;
    std::cout << "Popped element is: " << s.pop() << std::endl;
    std::cout << "Top element is now: " << s.peek() << std::endl;
    return 0;
}

File: stack.py
class Node:
    def __init__(self, data=None):
        self.data = data
        self.next = None

class Stack:
    def __init__(self):
        self.top = None
        self.size = 0

    def push(self, data):
        new_node = Node(data)
        new_node.next = self.top
        self.top = new_node
        self.size += 1

    def pop(self):
        if self.size == 0:
            return None
        temp = self.top
        self.top = self.top.next
        self.size -= 1
        return temp.data
 
stack = Stack()
 
stack.push(10)
stack.push(20)
stack.push(30)
 
print(stack.pop())  
print(stack.pop()) 
print(stack.pop())  
print(stack.pop())   
